src/Makefile

```diff
@@ -44,8 +44,8 @@ SRCS := \
 	jltypes gf typemap smallintset ast builtins module interpreter symbol \
 	dlload sys init task array genericmemory staticdata toplevel jl_uv datatype \
 	simplevector runtime_intrinsics precompile jloptions mtarraylist \
-	threading scheduler stackwalk gc gc-debug gc-pages gc-stacks gc-alloc-profiler gc-page-profiler method \
-	jlapi signal-handling safepoint timing subtype rtutils gc-heap-snapshot \
+	threading scheduler stackwalk method gc-alloc-profiler gc-heap-snapshot \
```

Comment:
The allocation profiler and heap snapshot are also stock GC-specific I believe?

Reply:
at least the heap snapshot definitely is.
I _think_ the allocation profiler is GC-agnostic, but i'm interested if that's not the case.

Reply:
You need to be careful to distinguish between the API (and its associated implementation) that's exposed by the heap snapshot, and code that uses this API (the stock GC in this case).

The heap snapshot file just implements the snapshot machinery and exposes an API that can be used by any GC implementation.

Because of that, I'd classify it as GC-agnostic.

Reply:
Something analogous holds for the allocation profiler IMO.

Reply:
We had a discussion about this a few weeks ago.

Conclusion was to move the heap snapshot into the library, but keep the allocation profiler out of it.